scripts/gen-universe.py

```diff
@@ -499,5 +495,77 @@ def create_index_entry(packages):
     return entry
 
 
+def v3_to_v2_package(v3_package):
+    """Converts a v3 package to a v2 package
+
+    :param v3_package: a v3 package
+    :type v3_package: dict
+    :return: a v2 package
+    :rtype: dict
+    """
+    package = copy.deepcopy(v3_package)
+
+    package.pop('minDcosReleaseVersion', None)
+    package['packagingVersion'] = "2.0"
+    resource = package.get('resource', None)
+    if resource:
+        cli = resource.pop('cli', None)
+        if cli and 'command' not in package:
+            print(('WARNING: Removing binary CLI from ({}, {}) without a '
+                  'Python CLI').format(package['name'], package['version']))
+
+    return package
+
+
+def v4_to_v3_package(v4_package):
+    """Converts a v4 package to a v3 package
+
+    :param v4_package: a v3 package
+    :type v4_package: dict
+    :return: a v3 package
+    :rtype: dict
+    """
+    package = copy.deepcopy(v4_package)
+    package.pop('upgradesFrom', None)
+    package.pop('downgradesTo', None)
+    return package
+
+
+def downgrade_package_to_v2(package):
+    """Converts a v4 or v3 package to a v2 package. If given a v2
+    package, it creates a deep copy but does not modify it. It does not
+    modify the original package.
+
+    :param package: v4, v3, or v2 package
+    :type package: dict
+    :return: a v2 package
+    :rtyte: dict
+    """
+    packaging_version = package.get("packagingVersion")
+    if packaging_version == "2.0":
+        return copy.deepcopy(package)
+    elif packaging_version == "3.0":
+        return v3_to_v2_package(package)
+    else:
+        return v3_to_v2_package(v4_to_v3_package(package))
+
+
+def downgrade_package_to_v3(package):
```

Comment:
That name is a little misleading since the function does not convert v2 packages to v3 packages.